packages/cli/src/executions/active-execution.service.ts

```diff
@@ -0,0 +1,143 @@
+import { Service } from 'typedi';
+import { ActiveExecutions } from '@/ActiveExecutions';
+import { Logger } from '@/Logger';
+import { Queue } from '@/Queue';
+import { WaitTracker } from '@/WaitTracker';
+import { ExecutionRepository } from '@db/repositories/execution.repository';
+import { getStatusUsingPreviousExecutionStatusMethod } from '@/executions/executionHelpers';
+
+import type { ExecutionSummary } from 'n8n-workflow';
+import type { IExecutionBase, IExecutionsCurrentSummary, IExecutionsStopData } from '@/Interfaces';
+import type { GetManyActiveFilter } from './execution.types';
+
+@Service()
+export class ActiveExecutionService {
+	constructor(
+		private readonly logger: Logger,
+		private readonly queue: Queue,
+		private readonly activeExecutions: ActiveExecutions,
+		private readonly executionRepository: ExecutionRepository,
+		private readonly waitTracker: WaitTracker,
+	) {}
+
+	async findOne(executionId: string, accessibleWorkflowIds: string[]) {
+		return await this.executionRepository.findIfAccessible(executionId, accessibleWorkflowIds);
+	}
+
+	private toSummary(execution: IExecutionsCurrentSummary | IExecutionBase): ExecutionSummary {
+		return {
+			id: execution.id,
+			workflowId: execution.workflowId ?? '',
+			mode: execution.mode,
+			retryOf: execution.retryOf !== null ? execution.retryOf : undefined,
+			startedAt: new Date(execution.startedAt),
+			status: execution.status,
+			stoppedAt: 'stoppedAt' in execution ? execution.stoppedAt : undefined,
+		};
+	}
+
+	// ----------------------------------
+	//           regular mode
+	// ----------------------------------
+
+	async findManyInRegularMode(
```

Comment:
We don't need this method anymore, neither the`findManyInQueueMode` below.

We can simply run `select * from execution_entity where status = 'running'` and it should return everything to us.

This is one of the largest benefits from the new `status` column. We used to run all those crazy queries simply because we could not infer what was running from the database alone.

Reply:
I was hesitant to make this change because executions are so undertested that I might end up inadvertently breaking something. I take it it's safe to go ahead with this then.